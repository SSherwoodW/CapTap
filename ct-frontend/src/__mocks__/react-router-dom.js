import { vi } from "vitest";

export * from "react-router-dom";

export const useParams = vi.fn();
// .. add more mocks if needed
