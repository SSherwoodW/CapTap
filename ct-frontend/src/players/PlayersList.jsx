import React, { useState, useEffect } from "react";
import Search from "../common/SearchForm";
import CapTapApi from "../../../api";
import PlayerCard from "../players/PlayerCard";
import LoadingSpinner from "../common/LoadingSpinner";

/** Show page with list of players.
 *
 * On mount, loads players from API.
 * Re-loads filtered players on submit from search form.
 *
 * PlayersList -> PlayerCard -> PlayerDetail
 *
 * This is routed to at /players
 */

function PlayersList() {
  console.debug("PlayersList");

  const [players, setPlayers] = useState(null);

  useEffect(function getAllPlayersOnMount() {
    console.debug("PlayersList useEffect getAllPlayersOnMount");
      search();
  }, []);

  /** Triggered by search form submit; reloads players. */
  async function search(name) {
    let players = await CapTapApi.getPlayers(name);
      setPlayers(players);
  }
    

  if (!players) return <LoadingSpinner />;
  

  return (
      <div className="PlayersList mx-auto max-w-7xl px-4 sm:px-6 lg:px-8">
          <Search searchFor={search} />
          {players.length
              ? (
                  <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4">
              {players.map(p => (
                  <PlayerCard
                      name={p.name}
                      id={p.id}
                      key={p.name}
                  />
              ))}
          </div>
              ) : (
                  <p className="lead">Sorry, no results were found!</p>
              )}
      </div>
  );
}

export default PlayersList;