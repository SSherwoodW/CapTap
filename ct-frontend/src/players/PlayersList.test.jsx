import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { beforeEach, describe, it, vi, fetchMock } from 'vitest';
import CapTapApi from "../../../api";
import PlayersList from './PlayersList';

vi.mock("../../../api");
const playersData = {
            id: '1',
            fullname: 'LeBron James',
            name: 'Los Angeles Lakers' 
        };
vi.spyOn(CapTapApi, 'getPlayers').mockResolvedValueOnce(playersData);
        
describe('PlayersList component', () => {
    it('should render without crashing', () => {
        render(
        <MemoryRouter>
            <PlayersList />
        </MemoryRouter>
        );
    });

    it('should match the snapshot', () => {
        const { container } = render(
        <MemoryRouter>
            <PlayersList />
        </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
        });

    it('should render loading spinner when players are null', () => {
        const { getByTestId } = render(
        <MemoryRouter>
            <PlayersList />
        </MemoryRouter>
        );
        expect(getByTestId('loading-spinner')).toBeInTheDocument();
    });

    it('should render players when players are not null', async () => {
        vi.spyOn(CapTapApi, 'getPlayers').mockResolvedValueOnce([
                { id: 1, name: 'Player 1' },
                { id: 2, name: 'Player 2' },
            ]);

        const { findByText } = render(
        <MemoryRouter>
            <PlayersList />
        </MemoryRouter>
        );
        screen.debug();

        // Check if loading spinner disappears and players are rendered
        expect(await findByText('Player 1')).toBeInTheDocument();
        expect(await findByText('Player 2')).toBeInTheDocument();
    });

    // 
});
