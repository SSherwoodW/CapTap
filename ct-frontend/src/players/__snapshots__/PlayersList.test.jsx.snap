// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayersList component > should match the snapshot 1`] = `
<div>
  <div
    class="LoadingSpinner"
    data-testid="loading-spinner"
  >
    Loading ...
  </div>
</div>
`;
